### ISSUE TITLE
Collaborate with the PM WG

### ISSUE THREAD

**Issue by wesleytodd** [feross/funding#15]
Are you interested in collaborating with the Package Maintenance Working Group?  We have been working on a spec for how a module can document it's "support".  This means financially as well as what consumers can expect for issues and updates.  I think that what you are doing here is interesting (ultimately I might disagree with the approach but not the spirit in which it is done) and he goal of the WG spec is similar in sprit.  The progress is here: https://github.com/nodejs/package-maintenance/pull/220

One idea I had for this was that once package authors annotate their packages in a predictable way we could have a tool like `npx support` and have it look at your project for modules requesting support.  Maybe even build some payment infrastructure around making it easy to split funding among all the things you use.

This might be a way we can achieve similar goals as this package but avoid the advertising model (which I am actually surprised you like based on the types of projects you build 😛).  Even if you are not interested in participating in the WG on a regular basis, I would be interested in hearing your thoughts on what we are proposing there.

**Comment by feross** [comment]
> Are you interested in collaborating with the Package Maintenance Working Group? We have been working on a spec for how a module can document it's "support".

Yes! I've been meaning to listen in on one your weekly calls. When is the next one?

> One idea I had for this was that once package authors annotate their packages in a predictable way we could have a tool like `npx support` and have it look at your project for modules requesting support

Have you seen [`npx thanks`](https://github.com/feross/thanks)? It's similar to what you're describing!

![](https://github.com/feross/thanks/raw/master/img/example.gif)

Unfortunately, since it's opt-in it didn't move the needle at all. I think it directed a few hundred dollars to Sindre's Patreon. But that was it.

**Comment by kemitchell** [comment]
> Unfortunately, since it's opt-in it didn't move the needle at all.

I think that's the real lesson of experience here. No little experience now.

No matter what your specific approach, if you hide your plea off the hot path, most people simply won't become aware that they can or should help you. Providing that data in a structured way only matters for those who've already decided to _go looking_ for it, and mostly for the devs who'd otherwise be sent to do that research manually.

Advertising dodges that trap by forcing its issue. By definition, ads put themselves in attention where they otherwise wouldn't belong.

Licensing, as through @licensezero, is kind of a middle ground. Lots of people ignore licensing. But a goodly number of folks, especially those working for companies, labor under rules that _force_ them to pay attention to `LICENSE` and package metadata. When they have to go looking, they get the message.

**Comment by feross** [comment]
> This might be a way we can achieve similar goals as this package but avoid the advertising model (which I am actually surprised you like based on the types of projects you build 😛)

The idea of this package is to extract some (admittedly very minor) amount of value from open source consumers who contribute nothing to the commons. Folks who contribute to open source should have an easy way to turn this off since they're not the target audience.

I’m not a huge fan of ads. I run an ad blocker in my browser and I understand the sentiments of the folks who feel strongly. (There's a way for OSS contributors to opt out here: https://github.com/feross/funding/pull/9) But like @kemitchell says, advertising forces the issue in a way that seems productive to me.

The mindset shift that I’ve gone through here is that I don’t want to ask nicely for donations anymore. I started asking myself – what leverage do I as a maintainer actually have here? What can I do to force a change in the dynamic?

**Comment by kemitchell** [comment]
> The mindset shift that I’ve gone through here is that I don’t want to ask nicely for donations anymore.

I wish I weren't right there with you, but I am.

May there always be exceptions. But the rule I've seen, over and over again, is that donations don't add up to meaningful cash, even for very well known developers writing popular software for popular platforms. Reliable amounts don't offset the shift in expectations that often accompanies any mention of money around a project, so developers often feel they end up worse off, overall.

It's got to the point that I see advice toward donations-based models as actively _unhelpful_. I have no doubt that many folks mean well. But it doesn't play out that way.

**Comment by wesleytodd** [comment]
> When is the next one?

https://github.com/nodejs/package-maintenance/issues/242

> Have you seen npx thanks? It's similar to what you're describing!

I saw it back when you first published it but apparently completely forgotten.  I think this tool is exactly what I meant and it would be great if we could build in support for the new spec once it is finalized.

> Unfortunately, since it's opt-in it didn't move the needle at all. I think it directed a few hundred dollars to Sindre's Patreon. But that was it.

I think this will also be the case with the first iterations of shipping this in `package.json`.  That said, I think as people like you push it forward we can build up an ecosystem which understands the issues maintainers face and hopefully have better empathy than is being shown right now.

If the goal is the short game of monetizing the work being done, I am afraid the results will be less than desirable.  IMO advertising is undesirable as a model, and is a short sighted approach.  Hopefully this experiment will be a great way to push the conversation forward, but if that is what we land on as a community I think we have made a grave mistake.

I think about how the most successful "free" products which chose an advertising model have ended up (google, facebook, etc).  I HATE what these products have done.  While this model is clearly good at generating revenue, it means that for maintainers to generate revenue they need to prove to advertisers that it is worth it.  Which means incentivizing exploiting user privacy to provide better metrics to companies.  I know *you* might not do this, but if the money model is ads others will.

Responses to things written while I was writing this:

> the idea of this package is to extract some (admittedly very minor) amount of value from open source consumers who contribute nothing to the commons.

This is exactly what I was worried about as a short sighted approach.  Please don't take this as a criticism, I don't mean to be negative here.  I really want a GREAT solution and am in support of trying different ideas.  I want this conversation to continue and be productive, which means also taking criticism :)

> Licensing, as through @licensezero, is kind of a middle ground. Lots of people ignore licensing. But a goodly number of folks, especially those working for companies, labor under rules that force them to pay attention to LICENSE and package metadata.

I think the best and most lucrative solution is one where companies are forced to pay.  Unfortunately this have proven hard.  I think a middle ground for this is making it easy and valuable for companies to pay.  I have some ideas about this, but am not sure if they are good ideas 🤷‍♂️

> that donations don't add up to meaningful cash, even for very well known developers writing popular software for popular platforms.

While I think that is true now, if you look at models like Twitch and other similar platforms it is clear that there are donation/subscription models which can.  We need to shift the conversation to make this happen in a way like Twitch did for monetizing gaming content.  If we make it easy for companies and organizations to support projects and get something out of it (like premium support, training, etc) maybe this would be different?

**Comment by traverseda** [comment]
>The idea of this package is to extract some (admittedly very minor) amount of value from open source consumers who contribute nothing to the commons. Folks who contribute to open source should have an easy way to turn this off since they're not the target audience.

If someone isn't making money from free software, why would you expect them to pay? The people who I'd expect to pay would be companies that are making money using your product.

**Comment by wesleytodd** [comment]
> If someone isn't making money from free software, why would you expect them to pay? The people who I'd expect to pay would be companies that are making money using your product.

I completely agree. One of the "not sure if they are good ideas" I want to try is offering premium support specifically targeted at companies.  I think this would be a way to have companies be more likely to get involved.  I actually [posted about this on express](https://github.com/expressjs/discussions/issues/69) a while ago, just never got traction.
